### PR TITLE
Add ICurrencyConverter dependencies to Security objects

### PR DESCRIPTION
**Pull request by mchandschuh**
#### Description

The scope of this change is small but the breadth is large, which is why I've decided to submit this as a standalone PR so it doesn't create noise for future changes that involve more involved logic/refactoring.

This change adds two new `ICurrencyConverter` implementations, an `IdentityCurrencyConverter` (currently unused), and an `ErrorCurrencyConverter`. The identity implementation returns the input `CashAmount` untouched unless the specified amount is not in units of the account currency, in which case it throws. This is a sensible default value that can be used in situations where we're guaranteed to be working in units of the account currency. This type is purposefully not used in LEAN infrastructure but is provided in the inevitable case that it's required to maintain backwards compatibility. The `ErrorCurrencyConverter` is intended to be used everywhere we assume that it won't be invoked, as it's implementation throws an exception immediately on every call. This implementation has widespread use among the exist unit tests and also in certain contexts where we create temporary security objects, for example, history requests, that won't require it's services. If in the future they do require these conversion services, we'll need to properly pipe the required algorithm dependencies into those providers.

#### Related Issue

See #1418 - this is but a small step towards satisfying the goals set out in this issue.

#### Motivation and Context

A (conceptually) small change to embed the `ICurrencyConverter` dependency privately inside security objects as it was identified as a place that requires it (for example, `TotalCloseProfit()`). This is being PR'd separately to avoid noise in future changes that have impact on behavior and logic.

#### Requires Documentation Change

No.

#### How Has This Been Tested?

Wrote unit tests for new classes, as well as existing coverage via regression and existing unit tests.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality) [on the way to a feature]
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/Securities/ErrorCurrencyConverter.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    /// </summary>
+    public class ErrorCurrencyConverter : ICurrencyConverter
+    {
+        public static ICurrencyConverter Instance = new ErrorCurrencyConverter();
+
+        private ErrorCurrencyConverter()
+        {
+        }
+
+        public CashAmount ConvertToAccountCurrency(CashAmount cashAmount)
```
Comment (StefanoRaggi) [inline Common/Securities/ErrorCurrencyConverter.cs]: XML comments are missing here.

File: Common/Securities/ErrorCurrencyConverter.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+using System;
+
+namespace QuantConnect.Securities
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="ICurrencyConverter"/> for use in
+    /// tests that don't depend on this behavior.
+    /// </summary>
+    public class ErrorCurrencyConverter : ICurrencyConverter
```
Comment (StefanoRaggi) [inline Common/Securities/ErrorCurrencyConverter.cs]: Nice idea 👍

File: QuantConnect.Lean.sln.DotSettings [hunk 3]
```diff
@@ -1,3 +1,6 @@
-<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
     <s:String x:Key="/Default/CodeInspection/CSharpLanguageProject/LanguageLevel/@EntryValue">CSharp60</s:String>
+	<s:Boolean x:Key="/Default/CodeStyle/CodeFormatting/CSharpFormat/WRAP_AFTER_INVOCATION_LPAR/@EntryValue">True</s:Boolean>
+	<s:String x:Key="/Default/CodeStyle/CodeFormatting/CSharpFormat/WRAP_ARGUMENTS_STYLE/@EntryValue">CHOP_IF_LONG</s:String>
+	<s:Boolean x:Key="/Default/CodeStyle/CodeFormatting/CSharpFormat/WRAP_BEFORE_INVOCATION_RPAR/@EntryValue">True</s:Boolean>
```
Comment (StefanoRaggi) [inline QuantConnect.Lean.sln.DotSettings]: Love the new style settings ❤️

Comment (mchandschuh) [reply QuantConnect.Lean.sln.DotSettings]: yah! these were initially an accident, but then I realized we should all be using the same style settings anyways, so going to start porting my settings to the team-shared settings -- basically as I run into them, more coming to a `DotSettings` file near you! :)

File: Common/Securities/ErrorCurrencyConverter.cs (hunk 2 above)
Comment (mchandschuh) [reply Common/Securities/ErrorCurrencyConverter.cs]: thanks! great way to confirm we _don't_ depend on it, super useful in isolating tests that need it and those that dont

File: Common/Securities/ErrorCurrencyConverter.cs (hunk 1 above)
Comment (mchandschuh) [reply Common/Securities/ErrorCurrencyConverter.cs]: will remedy. ty.

**Pull request merged**